src/js/game/systems/item_processor.js

```diff
@@ -196,14 +196,31 @@ export class ItemProcessorSystem extends GameSystemWithFilter {
 
                 const stackedDefinition = this.root.shapeDefinitionMgr.shapeActionStack(
                     lowerItem.definition,
-                    upperItem.definition
+                    upperItem.definition,
                 );
                 outItems.push({
                     item: new ShapeItem(stackedDefinition),
                 });
                 break;
             }
 
+            // UNSTACKER
+
+            case enumItemProcessorTypes.unstacker: {
+                const item = /** @type {ShapeItem} */ (itemsBySlot[0].item);
+                assert(item instanceof ShapeItem, "Input for unstack is not a shape");
+
+                const unstackedDefinitions = this.root.shapeDefinitionMgr.shapeActionUnstack(item.definition);
+
+                for (let i = 0; i < unstackedDefinitions.length; ++i) {
+                    outItems.push({
```

Comment:
As mentioned above - use `isEntirelyEmpty()` here and if so, don't emit on this slot